docs/en/routing/outlet-match-details.md

```diff
@@ -0,0 +1,71 @@
+# Using the Outlet MatchDetails
+
+For every `outlet` that is match on a route change `MatchDetails` are injected into the `Outlet` widget's `renderer` property. The `MatchDetails` object contains specific details for the matched outlet.
+
+* `queryParams: { [index: string]: string }`: The query params from the matched routing.
+* `params: { [index: string]: string }`: The path params from the matched routing.
+* `isExact(): boolean`: A function indicates if the outlet is an exact match for the path.
+* `isError(): boolean`: A function indicates if the outlet is an error match for the path.
+* `type: 'index' | 'partial' | 'error'`: The type of match for the route, either `index`, `partial` or `error`.
+* `router: RouterInterface`: The router instance which can used to create links and initiate route changes.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		children: [
+			{
+				path: 'details',
+				outlet: 'home-details'
+			}
+		]
+	}
+];
+```
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={(matchDetails) => {
+					const { params, queryParams, isExact, isError, router } = matchDetails;
+
+					const gotoHome = () => {
```

Comment:
This comment is still valid: https://github.com/dojo/framework/pull/286#discussion_r263755302